storage/innobase/handler/ha_innodb.cc

```diff
@@ -6290,6 +6294,17 @@ ha_innobase::close()
 {
 	DBUG_ENTER("ha_innobase::close");
 
+	auto& slice = export_vars.row_slices
+		[get_rnd_value()
```

Comment:
Should be worse on NUMA: now a node should fetch `array_elements(export_vars.row_slices)` cache lines!
If a transaction has more that one table, it will be sensed. I'd suggest to use getcpu for x86 instead if your glibc is fresh enough, at least for a benchmark purpose. Then we could improve the patch to support the rest of the platforms (sadly, arm64 is [still missing](https://elixir.bootlin.com/linux/v6.7/C/ident/__vdso_getcpu), however the [patch is ready](https://elixir.bootlin.com/linux/v6.7/C/ident/__vdso_getcpu) for as long as 3 years.

Reply:
Oh, you are right, `get_rnd_value()` primarily invokes `my_timer_cycles()`, which already by itself could be an expensive operation. If that returns 0, only then it will attempt an imitation of `getcpu()` by computing something from `pthread_self()`. In fact, the least significant 7 bits of that address may have no chance of uniquely identifying a thread. The address might be aligned to `CPU_LEVEL1_DCACHE_LINESIZE`, which could be 64 or 128 or an even larger power of 2.